### PR TITLE
Drop type specs in RecoTau and RecoVertex

### PR DESCRIPTION
**Pull request by NiharSaha**
#### PR description:
Drop type specs in Validation/RecoTau and Validation/RecoVertex python configuration files.

Central DQM migration campaign for drop type spces following
https://cms-sw.github.io/cms_coding_rules.html#6--packaging-rules-1
and update the safer syntax for existing parameters like: (1) drop type specifications where the original parameter exists. (2) using "clone" instead of "deepcopy" (3) move all parameters inside the clone

#### PR validation:

Tested in CMSSW_12_3_X via runTheMatrix.py -l limited -i all --ibeos

### CODE REVIEW

**Comment by qliphy** [comment]
please test

**Comment by jfernan2** [comment]
please abort

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
+1

File: Validation/RecoTau/python/RecoTauValidation_cff.py [hunk 1]
```diff
@@ -6,28 +6,28 @@
...
 tauValidationMiniAODRealData = tauValidationMiniAODZTT.clone(
-  RefCollection = cms.InputTag("CleanedPFJets"),
-  ExtensionName = cms.string('RealData')
+  RefCollection = "CleanedPFJets",
+  ExtensionName = 'RealData'
 )
 tauValidationMiniAODRealElectronsData = tauValidationMiniAODZTT.clone(
-  RefCollection = cms.InputTag("ElZLegs","theProbeLeg"),
+  RefCollection = "ElZLegs:theProbeLeg",
   ExtensionName = cms.string("RealElectronsData")
```
Comment (perrotta) [inline Validation/RecoTau/python/RecoTauValidation_cff.py:26]: Why not this one?

File: Validation/RecoVertex/python/HLTmultiPVvalidator_cff.py [hunk 2]
```diff
@@ -2,46 +2,46 @@
...
-hltMultiPVanalysis = vertexAnalysis.clone()
-hltMultiPVanalysis.do_generic_sim_plots  = False
-hltMultiPVanalysis.verbose               = cms.untracked.bool(False)
-hltMultiPVanalysis.root_folder           = cms.untracked.string("HLT/Vertexing/ValidationWRTsim")
-hltMultiPVanalysis.vertexRecoCollections = cms.VInputTag( )
-hltMultiPVanalysis.trackAssociatorMap    = cms.untracked.InputTag("trackingParticleRecoTrackAsssociation")
-hltMultiPVanalysis.vertexAssociator      = cms.untracked.InputTag("VertexAssociatorByPositionAndTracks")
-
+hltMultiPVanalysis = vertexAnalysis.clone(
+do_generic_sim_plots  = False,
```
Comment (perrotta) [inline Validation/RecoVertex/python/HLTmultiPVvalidator_cff.py:6]: Some tabbing would improve readability

File: Validation/RecoVertex/python/HLTmultiPVvalidator_cff.py [hunk 3]
```diff
@@ -2,46 +2,46 @@
...
+verbose               = False,
+root_folder           = "HLT/Vertexing/ValidationWRTsim",
+vertexRecoCollections = ("",),
```
Comment (perrotta) [inline Validation/RecoVertex/python/HLTmultiPVvalidator_cff.py:9]:
```suggestion
vertexRecoCollections = [""],
```

File: Validation/RecoVertex/python/HLTmultiPVvalidator_cff.py [hunk 4]
```diff
@@ -2,46 +2,46 @@
...
+trackAssociatorMap    = "trackingParticleRecoTrackAsssociation",
+vertexAssociator      = "VertexAssociatorByPositionAndTracks"
+)
 from Validation.RecoTrack.associators_cff import hltTrackAssociatorByHits, tpToHLTpixelTrackAssociation
 from SimTracker.VertexAssociation.VertexAssociatorByPositionAndTracks_cfi import VertexAssociatorByPositionAndTracks as _VertexAssociatorByPositionAndTracks
-vertexAssociatorByPositionAndTracks4pixelTracks = _VertexAssociatorByPositionAndTracks.clone()
-vertexAssociatorByPositionAndTracks4pixelTracks.trackAssociation = cms.InputTag("tpToHLTpixelTrackAssociation")
-
+vertexAssociatorByPositionAndTracks4pixelTracks = _VertexAssociatorByPositionAndTracks.clone(
+trackAssociation = "tpToHLTpixelTrackAssociation"
```
Comment (perrotta) [inline Validation/RecoVertex/python/HLTmultiPVvalidator_cff.py:16]: Please tab

File: Validation/RecoVertex/python/HLTmultiPVvalidator_cff.py [hunk 5]
```diff
@@ -2,46 +2,46 @@
...
+)
 tpToHLTpfMuonMergingTrackAssociation = tpToHLTpixelTrackAssociation.clone(
     label_tr = "hltPFMuonMerging"
 )
-vertexAssociatorByPositionAndTracks4pfMuonMergingTracks = _VertexAssociatorByPositionAndTracks.clone()
-vertexAssociatorByPositionAndTracks4pfMuonMergingTracks.trackAssociation = cms.InputTag("tpToHLTpfMuonMergingTrackAssociation")
-
+vertexAssociatorByPositionAndTracks4pfMuonMergingTracks = _VertexAssociatorByPositionAndTracks.clone(
+trackAssociation = "tpToHLTpfMuonMergingTrackAssociation"
```
Comment (perrotta) [inline Validation/RecoVertex/python/HLTmultiPVvalidator_cff.py:22]: please tab (etc: the same everywhere else)

File: Validation/RecoVertex/python/HLTmultiPVvalidator_cff.py [hunk 6]
```diff
@@ -2,46 +2,46 @@
...
-hltPixelPVanalysis.vertexAssociator      = cms.untracked.InputTag("vertexAssociatorByPositionAndTracks4pixelTracks")
-hltPixelPVanalysis.vertexRecoCollections = cms.VInputTag(
+hltPixelPVanalysis = hltMultiPVanalysis.clone(
+do_generic_sim_plots  = True,
+trackAssociatorMap    = "tpToHLTpixelTrackAssociation",
+vertexAssociator      = "vertexAssociatorByPositionAndTracks4pixelTracks",
+vertexRecoCollections = (
     "hltPixelVertices",
     "hltTrimmedPixelVertices",
 )
```
Comment (perrotta) [inline Validation/RecoVertex/python/HLTmultiPVvalidator_cff.py:33]:
```suggestion
vertexRecoCollections = ["hltPixelVertices", "hltTrimmedPixelVertices"],
```

File: Validation/RecoVertex/python/HLTmultiPVvalidator_cff.py [hunk 7]
```diff
@@ -2,46 +2,46 @@
...
-hltPVanalysis.vertexAssociator   = cms.untracked.InputTag("vertexAssociatorByPositionAndTracks4pfMuonMergingTracks")
-hltPVanalysis.vertexRecoCollections   = cms.VInputTag(
-    "hltVerticesPFFilter"
+hltPVanalysis = hltMultiPVanalysis.clone(
+trackAssociatorMap = "tpToHLTpfMuonMergingTrackAssociation",
+vertexAssociator   = "vertexAssociatorByPositionAndTracks4pfMuonMergingTracks",
+vertexRecoCollections   = (
+    "hltVerticesPFFilter",
 #    "hltFastPVPixelVertices"
 )
```
Comment (perrotta) [inline Validation/RecoVertex/python/HLTmultiPVvalidator_cff.py:43]:
```suggestion
vertexRecoCollections   = ["hltVerticesPFFilter"]   #"hltFastPVPixelVertices"
```
(or, even better, remove the commented out part if not needed any more)

File: Validation/RecoVertex/python/PrimaryVertexAnalyzer4PUSlimmed_cfi.py [hunk 8]
```diff
@@ -6,12 +6,12 @@
...
 )
 
-selectedOfflinePrimaryVerticesWithBS = selectedOfflinePrimaryVertices.clone()
-selectedOfflinePrimaryVerticesWithBS.src = cms.InputTag('offlinePrimaryVerticesWithBS')
-
-selectedPixelVertices = selectedOfflinePrimaryVertices.clone()
-selectedPixelVertices.src = cms.InputTag('pixelVertices')
-
+selectedOfflinePrimaryVerticesWithBS = selectedOfflinePrimaryVertices.clone(
+src = 'offlinePrimaryVerticesWithBS'
```
Comment (perrotta) [inline Validation/RecoVertex/python/PrimaryVertexAnalyzer4PUSlimmed_cfi.py:10]: tab

File: Validation/RecoVertex/python/PrimaryVertexAnalyzer4PUSlimmed_cfi.py [hunk 9]
```diff
@@ -6,12 +6,12 @@
...
+)
+selectedPixelVertices = selectedOfflinePrimaryVertices.clone(
+src = 'pixelVertices'
```
Comment (perrotta) [inline Validation/RecoVertex/python/PrimaryVertexAnalyzer4PUSlimmed_cfi.py:13]: tab

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
+1

**Comment by perrotta** [comment]
+1

**Pull request merged**